apps/passport-server/src/services/telegramService.ts

```diff
@@ -991,19 +995,54 @@ export class TelegramService {
   }
 
   /**
-   * Verify that a PCD relates to an event, and that the event has an
-   * associated chat. If so, invite the user to the chat and record them
-   * for later approval when they request to join.
-   *
    * This is called from the /telegram/verify route.
    */
   public async handleVerification(
-    serializedZKEdDSATicket: string,
+    serializedPCD: string,
     telegramUserId: number,
     telegramChatId: string,
     telegramUsername?: string
   ): Promise<void> {
     return traced("telegram", "handleVerification", async (span) => {
+      span?.setAttribute("chatId", telegramChatId);
+      const chat = await getGroupChat(this.authBot.api, telegramChatId);
+      span?.setAttribute("chatTitle", chat.title);
+
+      const parsed = JSON.parse(serializedPCD) as SerializedPCD;
+      if (parsed.type == ZKEdDSAEventTicketPCDTypeName) {
+        await this.handleTicketVerification(
+          serializedPCD,
+          telegramUserId,
+          chat.id,
+          telegramUsername
+        );
+      } else if (parsed.type == ZKEdDSAFrogPCDTypeName) {
+        await handleFrogVerification(
+          this.context.dbPool,
+          serializedPCD,
+          telegramUserId,
+          chat.id,
+          telegramUsername
+        );
+      }
```

Comment:
maybe another `else` here that throws an error? otherwise someone could send a random PCD and still get the link.